OpenRA.Mods.Common/Activities/Rearm.cs

```diff
@@ -17,44 +17,60 @@ namespace OpenRA.Mods.Common.Activities
 {
 	public class Rearm : Activity
 	{
-		readonly LimitedAmmo limitedAmmo;
 		int ticksPerPip = 25 * 2;
 		int remainingTicks = 25 * 2;
 		string sound = null;
 
 		public Rearm(Actor self, string sound = null)
 		{
-			limitedAmmo = self.TraitOrDefault<LimitedAmmo>();
+			var limitedAmmo = self.TraitsImplementing<LimitedAmmo>();
 			if (limitedAmmo != null)
-				ticksPerPip = limitedAmmo.ReloadTimePerAmmo();
-			remainingTicks = ticksPerPip;
-			this.sound = sound;
+			{
+				foreach (var la in self.TraitsImplementing<LimitedAmmo>())
+				{
+					ticksPerPip = la.ReloadTimePerAmmo();
+					remainingTicks = ticksPerPip;
+					this.sound = sound;
```

Comment:
I'm thinking you will need to either run a separate `Rearm` activity for each of `self.TraitsImplementing<LimitedAmmo>()`, or have a global `limitedAmmoTraits` collection in `Rearm` and populate it in this loop, then iterate through it in `Tick()`.